File: Numerical Method/Programs/My Lab/bisec.cpp
#include <iostream>
#include <cmath>
using namespace std;

float f(float x)
{
  return pow(x,3)-x-1;
}

float bisec(float a , float b , int n)
{
  float mid , res;
  while (n != 0)
  {
    mid = (a + b) / 2;
    res = f(mid);
    if (res < 0) 
    {
      a = mid;
    }
    else if (res > 0)
    {
      b = mid;
    }
    else 
    {
      break;
    }
    n--;
  }
  return mid;
}

int main()
{
  float a = 1, b = 2;
  int n = (log(b-a)-log(0.00001))/log(2);
  float res = bisec(a , b , n);
  cout << res << endl;
  return 0;
}

File: Numerical Method/Programs/My Lab/bisec.m
f = @(x) x^3 - x - 1; % the given function
a = 1; % initail guesses
b = 2;
err = 10^-4; % error toloerence
fa = f(a);
fb = f(b);
if fa*fb >= 0 % checking if bisection method is feseble
  disp('function must have opposite signs')
else
  while abs(b-a) > err  % main claculation
    mid = (a+b)/2;
    res = f(mid);
    if res > 0
      b = mid;
    elseif res < 0
      a = mid;
    else
      break
    endif
  endwhile
  fprintf("result is %f \n", mid) % output
end
